tee_supplicant/src/ree_fs.c

```diff
@@ -21,12 +21,59 @@ LOG_MODULE_REGISTER(ree_fs);
 #define REE_FS_PATHLEN sizeof(REE_FS_MP)
 #define REE_FS_PATH_MAX (PATH_MAX + REE_FS_PATHLEN)
 
+static int internal_open(char *path, fs_mode_t flags)
+{
+	int fd, rc;
+	struct fs_file_t *file;
+
+	file = k_malloc(sizeof(*file));
+	if (!file) {
+		return -ENOMEM;
+	}
+
+	fs_file_t_init(file);
+	fd = z_alloc_fd(file, NULL);
+	if (fd < 0) {
+		rc = TEEC_ERROR_GENERIC;
+		goto free;
+	}
+
+	rc = fs_open(file, path, flags);
+	if (rc < 0) {
+		goto free;
+	}
+	return fd;
+free:
+	if (fd >= 0) {
+		z_free_fd(fd);
+	}
+	k_free(file);
+	return rc;
+}
+
+static char *dname(char *path)
```

Comment:
This will not handle some cases like `//////` or `/a/../b` or `/./`

You need to either prove that those cases are impossible or account for all of them